### DISCUSSION TITLE
Internal framework using React and Payload CMS

### DISCUSSION THREAD

**Discussion by joshua-v-dev** [vikejs/vike#1483] [Discussion] [answered]
I am working on building an internal framework. I could use some help figuring out how to use Vike, which is partially pre-rendered, in a Turborepo alongside PayloadCMS as the backend, and Supabase as my database. I have my UI and utils packages being used as turborepo packages along with an assets loader package alongside the tailwind, eslint and typescript configs in packages directory. I have the PayloadCMS app and React frontend in separate directories within the apps folder. I am having trouble synchronizing the frontend with backend within this monorepo. If this is something anyone could help with I would appreciate it and can add whatever extra context or code that will help us figure this out. I am new to posting questions on here so I apologize if my format is incorrect.

**Comment by brillout** [comment]
The plan is to eventually write a lot of docs around building your own framework. But it isn't the highest priority right now, so it's going to take some time.

Until then, the expectation is that the user figures it on his own. If it's an option for you, you can [sponsor](https://github.com/sponsors/vikejs) and get help from us. We've been doing that with a couple of Vike sponsors who are building internal company frameworks.

**Reply by joshua-v-dev** [reply]
Thank you for taking the time to answer I know you are very busy. I am now a sponsor. I will increase my donation amount monthly as I am able. I am wanting to use this Vike project as my main portfolio piece I can use to get a job as a developer. I wish I had more to give now I apologize. I can manage most of the internal framework part myself. The main thing I am having trouble with right now is how to use my Vike frontend as the frontend for my PayoadCMS backend, which is running through express. I will do more research today, and see if I can figure it out. I have been struggling for 2 weeks trying to figure this out. Since I know you are so busy I will curate a more specific code example and targeted question so I am not asking you to take so much time. Again, thank you very much for your time. This Vike project you have created has changed my life for the better, and I just wanted to tell you that your project is prolific to say the least.

**Reply by brillout** [reply, marked as answer]
Thank you & have fun exploring.
